src/account-multicall/LlamaAccountMulticallFactory.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {LlamaAccountMulticallExtension} from "src/account-multicall/LlamaAccountMulticallExtension.sol";
+import {LlamaAccountMulticallGuard} from "src/account-multicall/LlamaAccountMulticallGuard.sol";
+import {LlamaAccountMulticallStorage} from "src/account-multicall/LlamaAccountMulticallStorage.sol";
+
+/// @title LlamaAccountMulticallFactory
+/// @author Llama (devsdosomething@llama.xyz)
+/// @notice This contract enables Llama instances to deploy an account multicall module.
+contract LlamaAccountMulticallFactory {
+  /// @dev Configuration of new Llama account multicall module.
+  struct LlamaAccountMulticallConfig {
+    address llamaExecutor; // The address of the Llama executor.
+    uint256 nonce; // The nonce of the new account multicall module.
+    LlamaAccountMulticallStorage.TargetSelectorAuthorization[] data; // The target-selectors to authorize.
+  }
+
+  /// @dev Emitted when a new Llama account multicall module is created.
+  event LlamaAccountMulticallModuleCreated(
+    address indexed deployer,
+    address indexed llamaExecutor,
+    uint256 nonce,
+    address accountMulticallGuard,
+    address accountMulticallExtension,
+    address accountMulticallStorage,
+    uint256 chainId
+  );
+
+  /// @notice Deploys a new Llama account multicall module.
+  /// @param accountMulticallConfig The configuration of the new Llama account multicall module.
+  /// @return accountMulticallGuard The deployed account multicall guard.
+  /// @return accountMulticallExtension The deployed account multicall extension.
+  /// @return accountMulticallStorage The deployed account multicall storage.
+  function deploy(LlamaAccountMulticallConfig memory accountMulticallConfig)
+    external
+    returns (
+      LlamaAccountMulticallGuard accountMulticallGuard,
+      LlamaAccountMulticallExtension accountMulticallExtension,
+      LlamaAccountMulticallStorage accountMulticallStorage
+    )
+  {
+    bytes32 salt =
+      keccak256(abi.encodePacked(msg.sender, accountMulticallConfig.llamaExecutor, accountMulticallConfig.nonce));
+
+    // Deploy and initialize account multicall storage.
+    accountMulticallStorage = new LlamaAccountMulticallStorage{salt: salt}(accountMulticallConfig.llamaExecutor);
+    accountMulticallStorage.initializeAuthorizedTargetSelectors(accountMulticallConfig.data);
```

Comment:
We have a separate `initializeAuthorizedTargetSelectors` (Protected by an `initialized` bool flag) so that we can have deterministic deploys. 

If we don't do it like this, the deployed address will also depend on the initial target-selector config to get the same address (Since `CREATE2` also depends on constructor args).

https://docs.soliditylang.org/en/latest/control-structures.html#salted-contract-creations-create2